### PR TITLE
Listview

### PR DESCRIPTION
**Pull request by pravagar**
List view page

-Added a reducer to hold states. Now there is one reducer(root reducer ) and state only holds samples.
- The page is having a fixed header and side bar.
- List view pane has a non scrollable fixed header.
- The link from list view to detail page works, but detail page is not styled and action are not proper.

### CODE REVIEW

File: ServerlessLibraryAPI/ClientApp/src/App.js [hunk 1]
```diff
@@ -1,42 +1,53 @@
...
-    this.getCurrentSample = this.getCurrentSample.bind(this)
-}
-
-getCurrentSample(id)
-{
-  return this.state.samples.filter(s=>s.title===id);
-}
-
+  componentDidMount() {
+    fetch("https://www.serverlesslibrary.net/api/Library")
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/App.js]: can you please move this code into a separate service, eg libraryService ?
getAll and getById (which is being used in detail view) can go into the service

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  },
+  textContent: {
+    padding: "15px 10px"
+  }
+});
+
+class DetailView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js]: do you need to have state here? can it just be props or some local data?

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  handleDeplayClick() {
+    var url =
+      "https://portal.azure.com/#create/Microsoft.Template/uri/" +
+      encodeURIComponent(this.state.sample.template);
+    window.open(url);
+  }
+
+  handleOpenInVSCodeClick() {
+    var url =
+      "vscode://vscode.git/clone?url=" +
```
[the remark this answers is not in the document]
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js]: This was unnecessary. will remove

File: ServerlessLibraryAPI/ClientApp/src/components/ItemTags/ItemTags.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import "./ItemTags.css";
+
+class ItemTags extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
[the remark this answers is not in the document]
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/ItemTags/ItemTags.js]: right

File: ServerlessLibraryAPI/ClientApp/src/components/ContentHeader/ContentHeader.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { withRouter } from "react-router-dom";
+import { SearchBox, Dropdown, Icon } from "office-ui-fabric-react/lib/index";
```
[the remark this answers is not in the document]
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/ContentHeader/ContentHeader.js]: never tried that. Just realized that it works

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+    }
+  }
+
+  handleLinkClick(pivotItem, ev) {
+    this.setState({
+      selectedKey: pivotItem.props.itemKey
+    });
+
+    if (
+      pivotItem.props.itemKey === "armtemplate" &&
```
[the remark this answers is not in the document]
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/DetailView.js]: right. The work is in progress

**Pull request merged**